xmake/modules/private/action/run/runenvs.lua

```diff
@@ -66,7 +66,25 @@ function _make_runpath_on_windows(target)
     return pathenv
 end
 
-function main(target)
+-- flatten envs ({PATH = {"A", "B"}} => {PATH = "A;B"})
+function _flatten_envs(envs)
+    local flatten_envs = {}
+    for name, values in pairs(envs) do
+        flatten_envs[name] = path.joinenv(values)
+    end
+    return flatten_envs
+end
+
+-- join addenvs and setenvs in a common envs table
+function join(addenvs, setenvs)
+    local envs = os.joinenvs(addenvs and _flatten_envs(addenvs) or {})
+    if setenvs then
+        table.join2(envs, _flatten_envs(setenvs))
```

Comment:
I think it should be 

```lua
function join(addenvs, setenvs)
    return os.joinenvs(addenvs and _flatten_envs(addenvs) or {}, setenvs and _flatten_envs(setenvs) or {})
end
```

Reply:
This won't take current envs into account, and will replace envs, as os.joinenvs use os.getenvs only if no table is passed

Reply:
> This won't take current envs into account, and will replace envs, as os.joinenvs use os.getenvs only if no table is passed

what about this? 
```lua
function join(addenvs, setenvs)
    local envs = os.joinenvs(addenvs and _flatten_envs(addenvs) or {})
    if setenvs then
        envs = os.joinenvs(envs, _flatten_envs(setenvs))
    end
    return envs
```

Reply:
That should work, however I'm away for a few days and won't be able to commit.